### ISSUE TITLE
Import on demand not only used on demand when solving types

### ISSUE THREAD

**Issue by mlangkabel** [javaparser/javasymbolsolver#180]
When solving symbols in parts of the JSS I have the following situation:
```java
import com.github.javaparser.ast.body.*;
import com.github.javaparser.symbolsolver.model.declarations.TypeDeclaration;
...
    TypeDeclaration x;
```

The type of `x` is solved as `com.github.javaparser.ast.body.TypeDeclaration` instead of `com.github.javaparser.symbolsolver.model.declarations.TypeDeclaration`. I think that the CompilationUnitContext just doesn't import the on-demand-imports on demand ;) I'll look into that.

**Comment by mlangkabel** [comment]
PR #182 fixes this issue :)

**Comment by mlangkabel** [comment]
fixed with merge of PR #182
